core/src/main/java/eu/fasten/core/merge/CGMerger.java

```diff
@@ -493,32 +495,51 @@ private void resolve(final Set<LongLongPair> edges,
      * Create a mapping from types and method signatures to callable IDs.
      *
      * @param dependenciesIds IDs of dependencies
-     * @param dbContext       DSL context
      * @param rocksDao        rocks DAO
      * @return a type dictionary
      */
     private Map<String, Map<String, LongSet>> createTypeDictionary(
-            final Set<Long> dependenciesIds, final DSLContext dbContext, final RocksDao rocksDao) {
+            final Set<Long> dependenciesIds, final RocksDao rocksDao) {
         final long startTime = System.currentTimeMillis();
         var result = new HashMap<String, Map<String, LongSet>>();
-
-        var callables = getCallables(dependenciesIds, rocksDao);
-
-        dbContext.select(Callables.CALLABLES.FASTEN_URI, Callables.CALLABLES.ID)
-                .from(Callables.CALLABLES)
-                .where(Callables.CALLABLES.ID.in(callables))
-                .fetch()
-                .forEach(callable -> {
-                    var node = new Node(FastenJavaURI.create(callable.value1()).decanonicalize());
-                    result.putIfAbsent(node.typeUri, new HashMap<>());
-                    var type = result.get(node.typeUri);
-                    var newestSet = new LongOpenHashSet();
-                    newestSet.add(callable.value2().longValue());
-                    type.merge(node.signature, newestSet, (old, newest) -> {
-                        old.addAll(newest);
-                        return old;
-                    });
-                });
+        int noCGCounter = 0, noMetadaCounter = 0;
+        for (Long dependencyId : dependenciesIds) {
+            DirectedGraph cg;
+            try {
```

Comment:
Maybe moving this to a helper method could help shorten this algorithm and make it more understandable